crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -833,16 +839,26 @@ impl SystemEnvironment {
 #[derive(Debug)]
 pub enum SitePackagesDiscoveryError {
     /// `site-packages` discovery failed because the provided path couldn't be canonicalized.
-    CanonicalizationError(SystemPathBuf, SysPrefixPathOrigin, io::Error),
+    CanonicalizationError(
+        SystemPathBuf,
+        SysPrefixPathOrigin,
+        io::Error,
+        Box<dyn System>,
```

Comment:
This is a bit annoying. I don't love it but the only alternative that I can think of is to a) use a thread local or b) have a global `System` somewhere (which breaks test isolation)

Reply:
Huh, yeah, I was initially surprised at these changes but I also don't see a better solution